### PR TITLE
Fix marking actions with invalid schedules as failed

### PR DESCRIPTION
**Pull request by thenbrent**
This PR reverts Prospress/action-scheduler#297 because it would mark the action as failed every time it was fetched, which included things like every time an admin page loads to display the action. The desired behaviour is to mark the action as failed once, when an attempt to run it is undertaken (and that attempt fails). I will introduce that in a separate PR, because it relies on some other improvements to how exceptions are handled when failing to fetch the action.

Because the code from Prospress/action-scheduler#297 has already been released, I've added a new `ActionScheduler_Store_Deprecated` class to safely deprecate the public and protected methods being reverted.

### CODE REVIEW

**Comment by rrennick** [comment]
This will create another merge conflict with #259 so, if we can, hold this one until after #259 is merged.

**Comment by thenbrent** [comment]
I've rebased this one on `version_3_0_0` now that #259 has been merged.

File: deprecated/ActionScheduler_Store_Deprecated.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+	/**
+	 * Mark an action that failed to fetch correctly as failed.
+	 *
+	 * @since 2.2.6
+	 *
+	 * @param int $action_id The ID of the action.
+	 */
+	public function mark_failed_fetch_action( $action_id ) {
+		_deprecated_function( __METHOD__, '3.0', 'ActionScheduler_Store::mark_failure()' );
```
Comment (rrennick) [inline deprecated/ActionScheduler_Store_Deprecated.php]:
```suggestion
		_deprecated_function( __METHOD__, '3.0.0', 'ActionScheduler_Store::mark_failure()' );
```

File: deprecated/ActionScheduler_Store_Deprecated.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+		self::$store->mark_failure( $action_id );
+	}
+
+	/**
+	 * Add base hooks
+	 *
+	 * @since 2.2.6
+	 */
+	protected static function hook() {
+		_deprecated_function( __METHOD__, '3.0' );
```
Comment (rrennick) [inline deprecated/ActionScheduler_Store_Deprecated.php]: 3.0.0

File: deprecated/ActionScheduler_Store_Deprecated.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	/**
+	 * Remove base hooks
+	 *
+	 * @since 2.2.6
+	 */
+	protected static function unhook() {
+		_deprecated_function( __METHOD__, '3.0' );
```
Comment (rrennick) [inline deprecated/ActionScheduler_Store_Deprecated.php]: 3.0.0

**Comment by thenbrent** [comment]
Thanks @rrennick! I've fixed up those issues.

**Pull request merged**